### PR TITLE
Remove BUNDLE_HOME And Move unset to entrypoint

### PR DESCRIPTION
**Pull request by nvick**
Move the unset env vars to the rails service entrypoint.

https://bundler.io/guides/bundler_docker_guide.html
https://bundler.io/v1.16/bundle_config.html

### CODE REVIEW

**Review by benjaminwood — APPROVED** [review]
Looks good.

One Nit:

https://github.com/hintmedia/railsdock/blob/237e13412ab176041083831f749c6a0d5b094dee/lib/railsdock/templates/install/default/Dockerfile#L64


Perhaps:
```
# Where Rubygems will look for gems
```

**Review by benjaminwood — APPROVED** [review]
Noice

File: lib/railsdock/templates/install/default/entrypoint.sh [hunk 1]
```diff
@@ -63,5 +63,8 @@ then
   if [[ "$4" = "s" || "$4" = "server" ]]; then rm -rf /app/tmp/pids/server.pid; fi
 fi
 
+unset BUNDLE_PATH
```
Comment (benjaminwood) [inline lib/railsdock/templates/install/default/entrypoint.sh:66]: @nvick after our experimenting today, do you believe that these `unsets's are ineffective?

Comment (nvick) [reply lib/railsdock/templates/install/default/entrypoint.sh:66]: Yeah, I believe they are ineffective.

Comment (nvick) [reply lib/railsdock/templates/install/default/entrypoint.sh:66]: Another PR removing `GEM_PATH` and the `unset`

**Pull request merged**